src/util/itt_notify.cpp

```diff
@@ -284,10 +295,17 @@ namespace hpx { namespace util { namespace itt
     task::task(domain const& domain, string_handle const& name)
       : domain_(domain), id_(0), sh_(name)
     {
-        id_ = HPX_ITT_MAKE_ID(domain_.domain_,
+        id_ = HPX_ITT_MAKE_ID(domain_.domain_.get(),
             reinterpret_cast<std::size_t>(sh_.handle_));
 
-        HPX_ITT_TASK_BEGIN_ID(domain_.domain_, id_, sh_.handle_);
+        HPX_ITT_TASK_BEGIN_ID(domain_.domain_.get(), id_, sh_.handle_);
+    }
+
+    task::~task()
+    {
+        HPX_ITT_TASK_END(domain_.domain_.get());
+
+        delete id_;
```

Comment:
Are you sure we can `delete` the memory? I have not gone back to the ITT docs, however AFAIR the ID is allocated using `malloc` (not using `new`).

Reply:
This one is allocated using `new` a bit further down. However, I'm not entire clear on what the difference between `itt_make_id` and `itt_id_create`. Maybe it should not be allocated manually in the first place?

Reply:
Ok, thanks for clarifying.